portal-api/src/v1/gef/controllers/companies.controller.js

```diff
@@ -0,0 +1,13 @@
+const externalApi = require('../../../external-api/api');
+
+const getCompanyByRegistrationNumber = async (req, res) => {
```

Comment:
Is there any nice way we could make this shared with the `v1` controller? I'm open to ideas about a nice pattern. 

The existing seperation of `gef` and `portal` makes it feel like we should maintain this function in this folder -- but I do feel it's not the best to repeat identical code. @abhi-markan is there a reason we care for having `gef` logic seperate from `portal` logic? I'm wondering if we either:
1. just keep this in it's existing `v1/controllers/companies.controller` and import it from there into our `gef routes`
2. keep it in it's existing `v1/controllers/companies.controller` and import it here -- then for gef we import it from here
3. create a `shared` folder in `portal-api` for things shared by `gef` and `portal` on our backend (not sure this is really helpful but might be)
4. keep two seperate definitions if theres an argument these are different on an abstract level (I really dont think they are)
5. other ideas that I've not thought of

Reply:
My preference would be for option 1 as I think that it is both the most simple and the clearest in terms of seeing where the controller actually comes from. I've done this for now but I'd be interested to hear @abhi-markan's take.